sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -458,7 +458,13 @@ export class BatchingReceiverLite {
           `${loggingPrefix} Drained, resolving receiveMessages() with ${brokeredMessages.length} messages.`
         );
 
-        resolve(brokeredMessages);
+        // NOTE: through rhea-promise most of our event handlers are made asynchronous by calling setTimeout(emit).
+        // However, a small set (*error and drain) execute immediately. This can lead to a situation where the logical
+        // ordering of events is correct but the execution order is incorrect because the events are not all getting
+        // put into the task queue the same way.
+        // So this call, while odd, just ensures that we resolve _after_ any already-queued onMessage handlers that may
+        // be waiting in the task queue.
+        setTimeout(() => resolve(brokeredMessages));
```

Comment:
Are you planning on adding a test for this?

Reply:
Yes, definitely :)

Reply:
Unit test added :)